imports/plugins/core/orders/client/templates/workflow/shippingInvoice.js

```diff
@@ -58,14 +59,33 @@ Template.coreOrderShippingInvoice.events({
 
     const state = instance.state;
     const order = state.get("order");
+    const orderTotal =
+      order.billing[0].invoice.subtotal
+      + order.billing[0].invoice.shipping
+      + order.billing[0].invoice.taxes;
     const discount = state.get("field-discount") || 0;
 
-    Meteor.call("orders/approvePayment", order, discount, (error) => {
-      if (error) {
-        // Show error
-        Logger.warn(error);
-      }
-    });
+    if (discount > orderTotal) {
+      Alerts.inline("Discount cannot be greater than original total price", "error", {
+        placement: "coreOrderShippingInvoice",
+        i18nKey: "order.invalidDiscount",
+        autoHide: 10000
+      });
+    } else {
+      Meteor.call("orders/approvePayment", order, discount, (error) => {
+        if (error) {
+          // Log & show error if client-side fails
```

Comment:
Correct, the server side method is not / should not be ever called if the discount > totalPrice, the validation is / should be all done client-side. 

The `// Log & show error if client-side fails` is there, and a second check is done once we get to the server-side, just in case somehow, someway `Meteor.call("orders/approvePayment"...` gets called without using the UI.